### ISSUE TITLE
Xlib w/o Xrender and screen scale has image buffer bug 

### ISSUE THREAD

**Issue by wcout** [fltk/fltk#159]
Drawing  transparency with XRrender disabled and screen scale set seems to have a buffer problem.
See attached test program containing comments.

[transp_test.tar.gz](https://github.com/fltk/fltk/files/5560829/transp_test.tar.gz)

Expected result:
![good](https://user-images.githubusercontent.com/9895976/99546301-96757980-29b6-11eb-950c-a8ee00404f6c.png)

Given result:

![wrong](https://user-images.githubusercontent.com/9895976/99546355-a2613b80-29b6-11eb-97a9-450ee96bc7f5.png)

**Comment by erco77** [comment]
Can replicate. On my system if I slide another window 1/2 over the black area, then pull it back away, the blue is shown in the revealed area. It also causes some little rectagular pixel artifacts (digital drit) in the other shapes, indicating there is some kind of drawing problem during redraws.

**Comment by erco77** [comment]
I can get the program to coredump (!) if I change Fl_Double_Window -> Fl_Window.

Seems to crash during `fl_read_image()` in the `alpha_blend()` function of `Xlib/Fl_Xlib_Graphics_Driver_image.cxx`, which I guess handles the blending of the incoming image to be drawn with whatever's already been drawn.

It does seem like when it's not crashing, fl_read_image() is the culprit in that it's not returning the same pixels that were drawn previously. I imagine with the scale this gets a bit complicated, mapping scaled pixels into and back out of the buffer.

One for @ManoloFLTK I'm thinking..

**Comment by erco77** [comment]
Dug a bit deeper still. For the Fl_Double_Window case where it doesn't crash, the call to `fl_read_image()` at the top of the `alpha_blend()` function is returning NULL, but since the image buffer it's supposed to fill is already allocated with `new`, it doesn't crash, it seems to just goes ahead and uses uninitialized memory as the background buffer, which is probably all zeroes (black).
I traced down into `fl_read_image()` to see why it's returning NULL; it calls `Fl_X11_Screen_Driver::read_win_rectangle()` which at some point calls `XGetImage()` which is supposed to return an image ptr but returns NULL. This seems to fall through that code, returning NULL from read_win_rectangle(), and thus to fl_read_image() which itself returns NULL that alpha_blend() is ignoring.

There appears to be an error handler surrounding the `XGetImage()` call (which does nothing), so I changed that error function to print the X windows error with this change:
```
   static int xgetimageerrhandler(Display *display, XErrorEvent *error) {
+    char s[128]; strcpy(s, "?"); XGetErrorText(display, error->error_code, s, 128-1);
+    printf("xgetimageerrhandler() returned X error '%s' (code %d)\n",  s, error->error_code);
     return 0;
   }
```
..and that seems to report:
`xgetimageerrhandler() returned X error 'BadMatch (invalid parameter attributes)' (code 8)`
I guess this means the `xid` isn't valid or something. I have to stop there I think.
